### DISCUSSION TITLE
Is it possible to store dynamic objects?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#9301] [Q&A]
I wish to store dynamic objects for workflow configuration. My intention is to allow the user to store dynamic objects which they can then choose to serialize into a workflow property during workflow setup. Is this possible?

**Comment by deanmarcussen** [comment]
You might need to provide a more detailed example of what you are trying to achieve.

I suspect what you want could be done with recipes, and `js[]` inside the recipe, but without an example, not sure this is a useful suggestion

**Reply by weirdyang** [reply]
thanks Dean.

I took a look at the custom settings module and What I want to achieve is exactly like that, except it needs to be able to be accessed during workflow creation.

For example a task which has a select input where the user can select the name of the custom settings and then set the object to a workflow property.

**Reply by weirdyang** [reply]
For the front end cshtml, which service is used to render this? ‘{{ Site.Properties }}

I am able to access {{ Site.Properties }} in the `log` task but when I try to use it to populate the url for the `http request task`, I get an error that it's an invalid uri.

edit; solved! need to use it like so  

{{ Site.Properties.NewSetting.NewSetting.EndPoint.Text | raw }} for the URL field as the expressionEvaluator uses a [urlendcoder](https://github.com/OrchardCMS/OrchardCore/blob/6820586d348e7aebf59644ea8e005d80697485b0/src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpRequestTask.cs#L170) but[ the `log task` doesn't.](https://github.com/OrchardCMS/OrchardCore/blob/6820586d348e7aebf59644ea8e005d80697485b0/src/OrchardCore.Modules/OrchardCore.Workflows/Activities/LogTask.cs#L48)
